### PR TITLE
add finish method to WandbLogger

### LINKED ISSUE #8614: 5-Fold with PyTorchLightning + Wandb seems to log to the same experiment
I am training 5-fold CV with PyTorch Lightning in a for loop. I am also logging all the results to wandb. I want wanbd to reinitalize the run after each fold, but it seems to continue with the same run and it logs all the results to the same run. I also tried passing kwargs in the WandbLogger as mentioned in the docs [here](https://pytorch-lightning.readthedocs.io/en/stable/extensions/generated/pytorch_lightning.loggers.WandbLogger.html#pytorch_lightning.loggers.WandbLogger.params.**kwargs), with no luck.
Here's a pseudo code of it:

```python
def run(fold):
    kwargs = {
        "reinit": True,
        "group": f"{CFG['exp_name']}"
    }
    wandb_logger = WandbLogger(project='<name>', 
                        entity='<entity>', 
                        config = CFG,
                        name=f"fold_{fold}",
                        **kwargs
            )
    trainer = Trainer(
        precision=16,
        gpus=1,
        fast_dev_run=False,
        callbacks = [checkpoint_callback],
        logger=wandb_logger,
        progress_bar_refresh_rate=1,
        max_epochs=2,
        log_every_n_steps=1

    )

    trainer.fit(
        lit_model,
        data_module
    )

if __name__ == "__main__":
    for fold in range(5):
        run(fold)
```

_Originally posted by @Gladiator07 in https://github.com/PyTorchLightning/pytorch-lightning/discussions/8572_

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Adds `WandbLogger.finish()` since WandbLogger by default operates with the global wandb.run object. 
Users who want to run several experiments in the same script must manually signal `wandb.finish()`. With the addition of this convenience method, they don't have to import wandb first just to do this.

Fixes #8614

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/loggers/wandb.py [hunk 1]
```diff
@@ -246,6 +246,14 @@ def finalize(self, status: str) -> None:
         if self._checkpoint_callback:
             self._scan_and_log_checkpoints(self._checkpoint_callback)
 
+    def finish(self) -> None:
```
Comment (ananthsub) [inline pytorch_lightning/loggers/wandb.py:249]: What's the difference between this, finalize, and close already on the logger base class?

Comment (awaelchli) [reply pytorch_lightning/loggers/wandb.py:249]: yes, that's the reason this PR is still in draft mode. I am not sure what I want to do here yet. Mainly I want to give a user control over ending an experiment manually.

Finalize: method of the loggers mentions this is for ending an experiment, but not actually all loggers implement it this way by closing their experiments completely. Finalize is called by the fit loop at the end of fit but most loggers will allow to continue logging for example during a subsequent .test() call. 

Close: This method exists, but never called by Lightning. Its purpose is unclear to me
